### PR TITLE
Do not fail on 422 error

### PR DESCRIPTION
**Pull request by gagoar**
## Description
Making sure 422 does not fail the entire action

## Motivation and Context
Having 422 is a common issue if the user running the rule is also somehow matching the reviewers added. 

## How Has This Been Tested?
unit tests

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/util/catchHandler.ts [hunk 1]
```diff
@@ -1,6 +1,16 @@
 import { Debugger } from 'debug';
+import { HttpErrors } from './constants';
+
+interface HttpError extends Error {
+  status: number;
+}
+
+export const catchHandler = (debug: Debugger) => (error: HttpError): Promise<unknown> => {
+  if (error.status < HttpErrors.SERVER_ERROR && error.status !== HttpErrors.RESOURCE_NOT_ACCESSIBLE) {
```
Comment (cyamonide) [inline src/util/catchHandler.ts:9]: A little uncomfortable that we're allowing any 4xx through as a non-fatal error. This includes 401 Unauthorized, 404 not founds, etc.

Comment (gagoar) [reply src/util/catchHandler.ts:9]: true. want me to consider only 403 ?

Comment (gagoar) [reply src/util/catchHandler.ts:9]: 404 could be a reviewer not found, not sure we wanna fail on that. it is a possible outcome that is not that important.

Comment (cyamonide) [reply src/util/catchHandler.ts:9]: I think we should do this on an allow-list basis. Right now we know that 422's are non-fatal, and in the future, if we encounter more stats codes that should be deemed non-fatal, we can add to that allowlist.

I don't think allowing all errors through and selectively picking errors as fatal is the right move here

Comment (cyamonide) [reply src/util/catchHandler.ts:9]: Reason being, it'll make it harder to catch false negatives (where no error was thrown but it should've been fatal)

Comment (gagoar) [reply src/util/catchHandler.ts:9]: sounds logical. I will do that

**Review by cyamonide — APPROVED** [review]

**Pull request merged**